quipucords/tests/api/reports/test_reports.py

```diff
@@ -290,3 +290,75 @@ def test_report_deployments_csv(django_client, deployment_with_logs: Deployments
     # Why 7? We expect two separate table headers in the file, one section marker,
     # three blank lines, and at least one line per table. Yes, this CSV is a mess. >:(
     assert len(tarball_deployments_csv_lines) >= 7
+
+
+def test_report_details_consistency(
+    django_client, deployment_with_logs: DeploymentsReport
+):
+    """Test report details response is driven by request headers and not cached."""
+    deployments_report = deployment_with_logs
+    report = deployments_report.report
+
+    for _ in range(2):
+        api_details_response = django_client.get(
+            reverse("v1:reports-details", args=(deployments_report.report.id,)),
+            headers={"Accept": "application/json"},
+        )
+        assert api_details_response.ok, api_details_response.text
+        api_details_json = api_details_response.json()
+        assert api_details_json["report_id"] == report.id
+        assert api_details_json["report_type"] == REPORT_TYPE_DETAILS
+        assert api_details_json["report_platform_id"] == str(report.report_platform_id)
+
+        details_csv_response = django_client.get(
+            reverse("v1:reports-details", args=(deployments_report.report.id,)),
+            headers={"Accept": "text/csv"},
+        )
+        assert details_csv_response.ok, details_csv_response.text
+
+        tarball_details_csv_lines = list(
+            csv.reader(details_csv_response.text.splitlines())
+        )
+        expected_csv_first_line = [
+            "Report ID",
+            "Report Type",
+            "Report Version",
+            "Report Platform ID",
+            "Number Sources",
+        ]
+        assert tarball_details_csv_lines[0] == expected_csv_first_line
+
+
+def test_report_deployments_consistency(
+    django_client, deployment_with_logs: DeploymentsReport
+):
+    """Test report deployments response is driven by request headers and not cached."""
+    deployments_report = deployment_with_logs
+    report = deployments_report.report
+
+    for _ in range(2):
```

Comment:
Same question as previous note about running this code twice. Just putting a comment here in case anything needs to change.